c/tskit/core.h

```diff
@@ -67,6 +69,33 @@ extern "C" {
 #define TSK_DBL_DECIMAL_DIG (DBL_DIG + 3)
 #endif
 
+/* We define a specific NAN value for default mutation time which indicates
+ * the time is unknown. We use a specific value so that if mutation time is set to
+ * a NAN from a computation we can reject it.
+ */
+#define TSK_MUTATION_UNKNOWN_TIME_HEX 0x7FF874736B697421ULL
+static inline double
+__tsk_nan_f(void)
+{
+    const union {
+        uint64_t i;
+        double f;
+    } nan_union = { TSK_MUTATION_UNKNOWN_TIME_HEX };
```

Comment:
This is safe, I think, but I had to look it up. Maybe it's a bit more explicit if we use this:
```c
const union {
￼        uint64_t i;
￼        double f;
￼    } nan_union = { .i=TSK_MUTATION_UNKNOWN_TIME_HEX };
```

Reply:
Fixed https://github.com/tskit-dev/tskit/pull/672/commits/ba7157c861afb08210407550bfc73a9bc84beeb7